controllers/machine_controller.go

```diff
@@ -115,13 +115,14 @@ func (r *MachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr e
 		}
 	}()
 
-	// Cluster might be nil as some providers might not require a cluster object
-	// for machine management.
+	// Reconcile and retrieve the Cluster object.
+	if m.Labels == nil {
+		m.Labels = make(map[string]string)
+	}
+	m.Labels[clusterv1.MachineClusterLabelName] = m.Spec.ClusterName
+
 	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, m.ObjectMeta)
```

Comment:
Should this be replaced with a different util function `util.GetClusterFromName` instead?